### PR TITLE
Pushpa_Agr_JS-Adagrams 

### PR DESCRIPTION
**Pull request by pushpaagr**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | I used my ruby knowledge of how loops and the logic work to navigate through javascript. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | Yes I noticed that when looking up some questions alot of info was outdated and had to really dig for the most current javascript info. 
What was a challenge you were able to overcome on this assignment? | It was difficult to try to debug the javascript code as we did not have access to binding.pry
What is your favorite thing about learning a new programming language? | I like that I am able to use logic to make the flow of the program run in similar way to ruby.
What is your least favorite thing about learning a new programming language? | It is difficult to debug sometimes
Do you have any recommendations on how we could improve this project for the next cohort? | More example/hw to go over syntax/logic

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | x, it's better to have commit messages that describe what the code changes were (such as "adds logic for scoring words") rather than "wave 2" etc
**Code Requirements** |
`drawLetters` method |  x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | x
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highestScoreFrom` pass | x
**Overall** |

Great work on this project! In this project you’ve taken some interesting logic and worked it into JavaScript syntax. I’m adding a few comments on places that this could be better and minor suggestions for improvement, but overall, you’ve done a great job of practicing syntax.

In general, my major syntax comments are:
You definitely missed a lot of semicolons and left a lot of `console.log`s in after submission. I like seeing your debugging skills, but it definitely clutters up the test output.

Otherwise, it looks good

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,160 @@
...
+      let i = 0
+      while (i < letters[letter] ) {
+        letterPool.push(letter)
+        i += 1
+      }
+    }
+
+    let k = 0
+    let playerHand = []
+    while (k < 10) {
```
Comment (tildeee) [inline src/adagrams.js:44]: instead of a `while` loop, it might make more sense to do a different loop that's more descriptive, like a `for` loop. That way you don't need to increment `k` at the end of the loop

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,160 @@
...
+      let random = Math.floor((Math.random() * (letterPool.length)) + 1);
```
Comment (tildeee) [inline src/adagrams.js:45]: Maybe instead of `random` there could be a better name for this, such as `randomIndex` or something?

File: src/adagrams.js [hunk 3]
```diff
@@ -1,7 +1,160 @@
...
+      playerHand.push(letterPool[random])
+      letterPool.splice(random, letterPool[random])
```
Comment (tildeee) [inline src/adagrams.js:47]: I don't think that the second parameter to `splice` should be `letterPool[random]`? According to the documentation, the first parameter is the starting index, which you've put as `random` which is correct... the second parameter is the number of elements to remove, which is saying that you should remove every instance of that letter. Don't you just want to remove one letter at a time?

File: src/adagrams.js [hunk 4]
```diff
@@ -1,7 +1,160 @@
...
+      k += 1
+    }
+
+    return playerHand
```
Comment (tildeee) [inline src/adagrams.js:51]: you're missing semicolons in every line of this function!

File: src/adagrams.js [hunk 5]
```diff
@@ -1,7 +1,160 @@
...
     // Implement this method for wave 1
   },
+
+  usesAvailableLetters(word, drawn) {
+    let letterArray = word.split("");
+    console.log(letterArray);
+    console.log(drawn);
```
Comment (tildeee) [inline src/adagrams.js:58]: Please remove all of the `console.log`s before submission

File: src/adagrams.js [hunk 6]
```diff
@@ -1,7 +1,160 @@
...
+        "J": 8,
+        "X": 8,
+        "Q": 10,
+        "Z": 10
+      }
+
+
+      let totalScore = 0;
+
+      let UpperCaseWord = word.toUpperCase();
```
Comment (tildeee) [inline src/adagrams.js:106]: you'll want to camelcase this to `upperCaseWord`

File: src/adagrams.js [hunk 7]
```diff
@@ -1,7 +1,160 @@
...
+      if (UpperCaseWord.length > 6) {
+        totalScore += 8;
+      }
+      console.log(totalScore);
+      return totalScore;
+
+    },
+
+
+    highestScoreFrom(words) {
```
Comment (tildeee) [inline src/adagrams.js:124]: watch your indentation!

File: src/adagrams.js [hunk 8]
```diff
@@ -1,7 +1,160 @@
...
+
+    const highest = {
+      word: '',
+      score: 0
+    }
+
+    for (let word of words) {
+
+      let scoreForWord = this.scoreWord(word)
```
Comment (tildeee) [inline src/adagrams.js:133]: missing semicolon

File: src/adagrams.js [hunk 9]
```diff
@@ -1,7 +1,160 @@
...
+          highest.score = scoreForWord;
+          highest.word = word;
+        }
+
+
+      }
+
+    }
+
+    return highest
```
Comment (tildeee) [inline src/adagrams.js:153]: missing semicolon
